include/zephyr/drivers/adc.h

```diff
@@ -707,6 +708,18 @@ __subsystem struct adc_driver_api {
 	uint16_t ref_internal;	/* mV */
 };
 
+/**
+ * @brief Verify that a device has an ADC driver API.
+ *
+ * @param dev a device pointer
+ *
+ * @return true if the device's API is located in the ADC API section.
+ */
+static inline bool device_has_adc_driver_api(const struct device *dev)
```

Comment:
-1 on using `device_` namespace out of `device_` header, maybe `adc_device_check` or something else that uses `adc_`.